ext/drutil/drutil.c

```diff
@@ -429,6 +429,32 @@ drutil_insert_get_mem_addr_arm(void *drcontext, instrlist_t *bb, instr_t *where,
 }
 #endif     /* X86/AARCHXX */
 
+static inline void
+native_cpuid(uint *eax, uint *ebx, uint *ecx, uint *edx)
+{
+#ifdef WINDOWS
+    /* XXX i#2946: support Windows. */
```

Comment:
This is very easy to do using intrinsics and IMHO we should just do it now to avoid adding holes in the API.  See e.g. clients/drcpusim/tests/cpuid.c.
```
            int output[4];
            __cpuidex(output, op, subop);
```

If it's not going to be there now the doxygen docs need to be updated stating that Windows behaves differently.  It seems easier to just implement it, since it's actually easier on Windows than on Linux.

Reply:
Done.